src/verify-request/verify-token.ts

```diff
@@ -25,9 +26,21 @@ export function verifyToken(routes: Routes, accessMode: AccessMode = DEFAULT_ACC
       const scopesChanged = !Shopify.Context.SCOPES.equals(session.scope);
 
       if (!scopesChanged && session.accessToken && (!session.expires || session.expires >= new Date())) {
-        ctx.cookies.set(TOP_LEVEL_OAUTH_COOKIE_NAME);
-        await next();
-        return;
+        try {
+          // make a request to make sure oauth has succeeded, retry otherwise
+          const client = new Shopify.Clients.Rest(session.shop, session.accessToken)
+          await client.get({ path: "metafields", query: {'limit': 1} }) 
+
+          ctx.cookies.set(TOP_LEVEL_OAUTH_COOKIE_NAME);
+          await next();
+          return;
+        } catch(e) {
+          if (e instanceof HttpResponseError && e.code == 401){
+              // only catch 401 errors
+          } else {
+            throw e
+          }
```

Comment:
```suggestion
          if (e instanceof HttpResponseError && e.code != 401) {
              // only throw non-401 errors
            throw e
          }
```

Reply:
For the lines just above this `try` block, there are identical lines in a PR by @carmelal 
https://github.com/Shopify/shopify-node-api/blob/597c870978c9b6cb9b187ea240e1841f0bfd062a/src/auth/session/session.ts#L33-L34

I'm wondering if this change should wait for that other one to be merged?

Reply:
I think Carmela's changes are updating this same place, so we just need to make sure they're merged properly.

Reply:
I also have a question about this block, which I think relates to Kevin's suggested change: 
Since there is nothing in the `if` block, won't this _not_ do anything with the error when it _is_ a `401`? Right now we only seem to `throw` in the `else`. Or is the intention for it to error silently in those cases (`401`)? Sorry if I'm missing something obvious here!

Reply:
The 'silent error' here means we proceed as normal when there is no session - i.e. if the session expired we go to `/auth`.

Reply:
Ah ok! Then this works for me! Sorry I misunderstood the intention here.